php-checks/src/main/java/org/sonar/php/checks/StringLiteralDuplicatedCheck.java

```diff
@@ -66,9 +102,10 @@ public void visitCompilationUnit(CompilationUnitTree tree) {
 
   private void finish() {
     for (Map.Entry<String, List<LiteralTree>> literalOccurrences : sameLiteralOccurrences.entrySet()) {
+      String value = literalOccurrences.getKey();
       List<LiteralTree> occurrences = literalOccurrences.getValue();
 
-      if (occurrences.size() >= threshold) {
+      if (occurrences.size() >= threshold && !ALLOWED_DUPLICATED_LITERALS.matcher(value).matches()) {
```

Comment:
- Before: if the regex matches, we don't add to the `sameLiteralOccurrences` map
- After: we add to the `sameLiteralOccurrences` map, and discard if the regex matches

We could still check if the regex matches before adding to the map, and caching the results in another map to avoid having a useless big `sameLiteralOccurrences` map in the heap. It would be pretty fast especially with JVM's string pooling, but I'm not sure it's worth the complexity, it would only help on very big specific files 🤔

Reply:
Both logic are interesting, they have different pros and cons.
The current logic sacrifices memory for time, while yours does the opposite with a good optimisation on the execution.
I would keep the current logic at least for now, as I'm not sure what is the best strategy on typical PHP code, but to be kept in mind if we see impact in the upcoming days on peach.